# Atividade A13 - Intervalos de Confianca

Este notebook apresenta a resolucao de tres exercicios sobre intervalos de confianca, com explicacoes passo a passo e observacoes para uma possivel defesa de autoria.

Os exercicios abordam:

1. intervalo de confianca de 95% para uma media, usando o Teorema do Limite Central;
2. intervalo de confianca de 95% para a proporcao de lampadas defeituosas;
3. intervalo de confianca de 99% para a proporcao de votos favoraveis no exemplo do candidato a senador XYMK?T.

A primeira resolucao usa os seguintes dados:

- media amostral: $\bar{x} = 31.3$
- desvio padrao populacional: $\sigma = 7.6$
- tamanho da amostra: $n = 40$
- nivel de confianca: 95%
- populacao original: nao normalmente distribuida

## Justificativa estatistica

Como a populacao nao e normalmente distribuida, nao podemos assumir diretamente normalidade para os dados individuais.

Mesmo assim, como o tamanho da amostra e $n = 40$, que e maior ou igual a 30, podemos aplicar o **Teorema do Limite Central**. Esse teorema afirma que, para amostras suficientemente grandes, a distribuicao da media amostral se aproxima de uma distribuicao normal, independentemente do formato da distribuicao original da populacao.

Como o desvio padrao populacional foi informado ($\sigma = 7.6$), usamos a distribuicao normal padrao, com valor critico:

$$
z_{0.025} = 1.96
$$

## Formula utilizada

O intervalo de confianca para a media, com desvio padrao populacional conhecido, e dado por:

$$
IC = \bar{x} \pm z \cdot \frac{\sigma}{\sqrt{n}}
$$

Onde:

- $\bar{x}$ e a media da amostra
- $z$ e o valor critico da distribuicao normal padrao
- $\sigma$ e o desvio padrao da populacao
- $n$ e o tamanho da amostra
- $\frac{\sigma}{\sqrt{n}}$ e o erro padrao da media

In [5]:
import math

media_amostral = 31.3
desvio_padrao_populacional = 7.6
n = 40
z_95 = 1.96

erro_padrao = desvio_padrao_populacional / math.sqrt(n)
margem_erro = z_95 * erro_padrao

limite_inferior = media_amostral - margem_erro
limite_superior = media_amostral + margem_erro

print("Exercicio 1 - Intervalo de confianca para a media")
print(f"Media amostral: {media_amostral:.4f}")
print(f"Desvio padrao populacional: {desvio_padrao_populacional:.4f}")
print(f"Tamanho da amostra: {n}")
print(f"Valor critico z para 95%: {z_95:.4f}")
print(f"Erro padrao da media: {erro_padrao:.4f}")
print(f"Margem de erro: {margem_erro:.4f}")
print(f"Intervalo de confianca de 95%: [{limite_inferior:.4f}, {limite_superior:.4f}]")

Exercicio 1 - Intervalo de confianca para a media
Media amostral: 31.3000
Desvio padrao populacional: 7.6000
Tamanho da amostra: 40
Valor critico z para 95%: 1.9600
Erro padrao da media: 1.2017
Margem de erro: 2.3553
Intervalo de confianca de 95%: [28.9447, 33.6553]


## Calculo passo a passo

Primeiro calculamos o erro padrao da media:

$$
SE = \frac{7.6}{\sqrt{40}}
$$

$$
SE \approx 1.2017
$$

Depois calculamos a margem de erro para 95% de confianca:

$$
ME = 1.96 \cdot 1.2017
$$

$$
ME \approx 2.3553
$$

Aplicando na media amostral:

$$
IC = 31.3 \pm 2.3553
$$

$$
IC \approx [28.9447, 33.6553]
$$

In [2]:
print("Resumo do resultado do Exercicio 1")
print(f"Erro padrao: {erro_padrao:.4f}")
print(f"Margem de erro: {margem_erro:.4f}")
print(f"Intervalo de confianca de 95%: [{limite_inferior:.4f}, {limite_superior:.4f}]")

Resumo do resultado do Exercicio 1
Erro padrao: 1.2017
Margem de erro: 2.3553
Intervalo de confianca de 95%: [28.9447, 33.6553]


## Resultado final

O intervalo de confianca de 95% para a media populacional e:

$$
\boxed{[28.9447,\ 33.6553]}
$$

Arredondando para duas casas decimais:

$$
\boxed{[28.94,\ 33.66]}
$$

Interpretacao: com 95% de confianca, estimamos que a verdadeira media populacional esteja entre **28.94** e **33.66**.

Para uma defesa de autoria, o ponto principal e explicar que a populacao original nao precisa ser normal neste caso porque a amostra tem tamanho 40. Pelo Teorema do Limite Central, a distribuicao da media amostral pode ser aproximada por uma normal. Como o desvio padrao populacional foi informado, usamos o valor critico $z = 1.96$, e nao a distribuicao t de Student.

# Exercicio 2 - Intervalo de Confianca para Proporcao

Uma amostra de 200 lampadas foi testada e 11 foram encontradas com defeito. O objetivo e calcular um intervalo de confianca de 95% para a proporcao populacional de lampadas defeituosas.

Dados do problema:

- tamanho da amostra: $n = 200$
- quantidade de lampadas defeituosas: $x = 11$
- proporcao amostral: $\hat{p} = \frac{11}{200} = 0.055$
- complemento: $\hat{q} = 1 - \hat{p} = 0.945$
- nivel de confianca: 95%
- valor critico: $z = 1.96$

## Justificativa estatistica

Este e um problema de intervalo de confianca para uma **proporcao**, pois cada lampada testada pode ser classificada em uma de duas categorias: defeituosa ou nao defeituosa.

Podemos usar a aproximacao normal para a proporcao porque as condicoes sao satisfeitas:

$$
n\hat{p} = 200 \cdot 0.055 = 11 \geq 5
$$

$$
n\hat{q} = 200 \cdot 0.945 = 189 \geq 5
$$

Como as duas quantidades sao maiores ou iguais a 5, a distribuicao da proporcao amostral pode ser aproximada pela distribuicao normal.

## Formula utilizada

O intervalo de confianca para uma proporcao e:

$$
IC = \hat{p} \pm z \cdot \sqrt{\frac{\hat{p}\hat{q}}{n}}
$$

Onde:

- $\hat{p}$ e a proporcao observada na amostra
- $\hat{q}$ e $1 - \hat{p}$
- $n$ e o tamanho da amostra
- $z$ e o valor critico para o nivel de confianca
- $\sqrt{\frac{\hat{p}\hat{q}}{n}}$ e o erro padrao da proporcao

In [3]:
lampadas_testadas = 200
lampadas_defeituosas = 11
z_95 = 1.96

proporcao_defeituosas = lampadas_defeituosas / lampadas_testadas
proporcao_nao_defeituosas = 1 - proporcao_defeituosas

erro_padrao_proporcao = math.sqrt(
    (proporcao_defeituosas * proporcao_nao_defeituosas) / lampadas_testadas
)
margem_erro_proporcao = z_95 * erro_padrao_proporcao

limite_inferior_proporcao = proporcao_defeituosas - margem_erro_proporcao
limite_superior_proporcao = proporcao_defeituosas + margem_erro_proporcao

print("Exercicio 2 - Intervalo de confianca para proporcao")
print(f"Lampadas testadas: {lampadas_testadas}")
print(f"Lampadas defeituosas: {lampadas_defeituosas}")
print(f"Proporcao amostral de defeituosas: {proporcao_defeituosas:.4f}")
print(f"Proporcao complementar: {proporcao_nao_defeituosas:.4f}")
print(f"Valor critico z para 95%: {z_95:.4f}")
print(f"Erro padrao da proporcao: {erro_padrao_proporcao:.4f}")
print(f"Margem de erro: {margem_erro_proporcao:.4f}")
print(f"Intervalo de confianca de 95%: [{limite_inferior_proporcao:.4f}, {limite_superior_proporcao:.4f}]")
print(f"Intervalo em porcentagem: [{limite_inferior_proporcao * 100:.2f}%, {limite_superior_proporcao * 100:.2f}%]")

Exercicio 2 - Intervalo de confianca para proporcao
Lampadas testadas: 200
Lampadas defeituosas: 11
Proporcao amostral de defeituosas: 0.0550
Proporcao complementar: 0.9450
Valor critico z para 95%: 1.9600
Erro padrao da proporcao: 0.0161
Margem de erro: 0.0316
Intervalo de confianca de 95%: [0.0234, 0.0866]
Intervalo em porcentagem: [2.34%, 8.66%]


## Calculo passo a passo

A proporcao amostral de lampadas defeituosas e:

$$
\hat{p} = \frac{11}{200} = 0.055
$$

O erro padrao da proporcao e:

$$
SE = \sqrt{\frac{0.055 \cdot 0.945}{200}}
$$

$$
SE \approx 0.0161
$$

A margem de erro para 95% de confianca e:

$$
ME = 1.96 \cdot 0.0161 \approx 0.0316
$$

Logo:

$$
IC = 0.055 \pm 0.0316
$$

$$
IC \approx [0.0234, 0.0866]
$$

In [4]:
print("Resumo do resultado do Exercicio 2")
print(f"Proporcao amostral: {proporcao_defeituosas:.4f}")
print(f"Erro padrao: {erro_padrao_proporcao:.4f}")
print(f"Margem de erro: {margem_erro_proporcao:.4f}")
print(
    "Intervalo de confianca de 95%: "
    f"[{limite_inferior_proporcao:.4f}, {limite_superior_proporcao:.4f}]"
)
print(
    "Em porcentagem: "
    f"[{limite_inferior_proporcao * 100:.2f}%, {limite_superior_proporcao * 100:.2f}%]"
)

Resumo do resultado do Exercicio 2
Proporcao amostral: 0.0550
Erro padrao: 0.0161
Margem de erro: 0.0316
Intervalo de confianca de 95%: [0.0234, 0.0866]
Em porcentagem: [2.34%, 8.66%]


## Resultado final do Exercicio 2

O intervalo de confianca de 95% para a proporcao de lampadas defeituosas e:

$$
\boxed{[0.0234,\ 0.0866]}
$$

Em porcentagem:

$$
\boxed{[2.34\%,\ 8.66\%]}
$$

Interpretacao: com 95% de confianca, estimamos que a proporcao real de lampadas defeituosas na populacao esteja entre **2.34%** e **8.66%**.

Para uma defesa de autoria, o ponto principal e explicar que este exercicio trabalha com proporcao, nao com media. Por isso usamos $\hat{p}$, $\hat{q}$ e o erro padrao da proporcao. A aproximacao normal e valida porque $n\hat{p}$ e $n\hat{q}$ sao maiores ou iguais a 5.

# Exercicio 3 - Intencao de Voto com 99% de Confianca

Neste exercicio usamos o exemplo do candidato a senador XYMK?T apresentado no material.

No exemplo original:

- foram entrevistados $n = 1000$ eleitores
- 550 disseram que votariam no candidato
- portanto, a proporcao amostral e $\hat{p} = 0.55$
- o complemento e $\hat{q} = 1 - 0.55 = 0.45$

Agora o candidato quer aumentar o nivel de confianca para 99%. De acordo com a tabela do material, para 99% usamos:

$$
z = 2.58
$$

## Formula utilizada

Como estamos estimando uma proporcao de votos favoraveis, usamos novamente o intervalo de confianca para proporcao:

$$
IC = \hat{p} \pm z \cdot \sqrt{\frac{\hat{p}\hat{q}}{n}}
$$

A diferenca em relacao ao exemplo de 95% e que o valor critico aumenta de $1.96$ para $2.58$. Por isso, o intervalo fica mais largo: maior confianca exige menor precisao.

In [5]:
eleitores_entrevistados = 1000
proporcao_votos_favoraveis = 0.55
proporcao_votos_nao_favoraveis = 1 - proporcao_votos_favoraveis
z_99 = 2.58

erro_padrao_votos = math.sqrt(
    (proporcao_votos_favoraveis * proporcao_votos_nao_favoraveis)
    / eleitores_entrevistados
)
margem_erro_votos_99 = z_99 * erro_padrao_votos

limite_inferior_votos_99 = proporcao_votos_favoraveis - margem_erro_votos_99
limite_superior_votos_99 = proporcao_votos_favoraveis + margem_erro_votos_99

print("Exercicio 3 - Intervalo de confianca para intencao de voto")
print(f"Eleitores entrevistados: {eleitores_entrevistados}")
print(f"Proporcao amostral de votos favoraveis: {proporcao_votos_favoraveis:.4f}")
print(f"Proporcao complementar: {proporcao_votos_nao_favoraveis:.4f}")
print(f"Valor critico z para 99%: {z_99:.4f}")
print(f"Erro padrao da proporcao: {erro_padrao_votos:.4f}")
print(f"Margem de erro para 99%: {margem_erro_votos_99:.4f}")
print(f"Intervalo de confianca de 99%: [{limite_inferior_votos_99:.4f}, {limite_superior_votos_99:.4f}]")
print(f"Intervalo em porcentagem: [{limite_inferior_votos_99 * 100:.2f}%, {limite_superior_votos_99 * 100:.2f}%]")

Exercicio 3 - Intervalo de confianca para intencao de voto
Eleitores entrevistados: 1000
Proporcao amostral de votos favoraveis: 0.5500
Proporcao complementar: 0.4500
Valor critico z para 99%: 2.5800
Erro padrao da proporcao: 0.0157
Margem de erro para 99%: 0.0406
Intervalo de confianca de 99%: [0.5094, 0.5906]
Intervalo em porcentagem: [50.94%, 59.06%]


## Calculo passo a passo

A proporcao amostral e:

$$
\hat{p} = \frac{550}{1000} = 0.55
$$

O erro padrao da proporcao e:

$$
SE = \sqrt{\frac{0.55 \cdot 0.45}{1000}}
$$

$$
SE \approx 0.0157
$$

A margem de erro para 99% de confianca e:

$$
ME = 2.58 \cdot 0.0157 \approx 0.0406
$$

Logo:

$$
IC = 0.55 \pm 0.0406
$$

$$
IC \approx [0.5094, 0.5906]
$$

In [6]:
print("Resumo do resultado do Exercicio 3")
print(f"Proporcao amostral: {proporcao_votos_favoraveis:.4f}")
print(f"Erro padrao: {erro_padrao_votos:.4f}")
print(f"Margem de erro para 99%: {margem_erro_votos_99:.4f}")
print(
    "Intervalo de confianca de 99%: "
    f"[{limite_inferior_votos_99:.4f}, {limite_superior_votos_99:.4f}]"
)
print(
    "Em porcentagem: "
    f"[{limite_inferior_votos_99 * 100:.2f}%, {limite_superior_votos_99 * 100:.2f}%]"
)

Resumo do resultado do Exercicio 3
Proporcao amostral: 0.5500
Erro padrao: 0.0157
Margem de erro para 99%: 0.0406
Intervalo de confianca de 99%: [0.5094, 0.5906]
Em porcentagem: [50.94%, 59.06%]


## Resultado final do Exercicio 3

Com 99% de confianca, a faixa em que estaria a proporcao de votos a favor do candidato e:

$$
\boxed{[0.5094,\ 0.5906]}
$$

Em porcentagem:

$$
\boxed{[50.94\%,\ 59.06\%]}
$$

Interpretacao: com 99% de confianca, estimamos que a proporcao real de votos a favor do candidato esteja entre **50.94%** e **59.06%**.

Para uma defesa de autoria, o ponto principal e explicar que o intervalo ficou maior do que o intervalo de 95% apresentado no exemplo original. Isso acontece porque aumentar o nivel de confianca de 95% para 99% exige usar um valor critico maior, aumentando a margem de erro.